include/CL/opencl.hpp

```diff
@@ -10253,6 +10342,530 @@ namespace compatibility {
+                (sync_point != NULL) ? &tmp_sync_point : NULL,
+                mutable_handle),
+            __COMMAND_COPY_IMAGE_KHR_ERR);
+
+        if (sync_point != NULL && error == CL_SUCCESS)
+            *sync_point = tmp_sync_point;
+
+        return error;
+    }
+
+    cl_int commandCopyImageToBuffer(const CommandQueue* command_queue,
+        const Image& src,
+        const Buffer& dst,
+        const array<size_type, 3>& src_origin,
+        const array<size_type, 3>& region,
+        size_type dst_offset,
+        const vector<cl_sync_point_khr>* sync_points_vec = NULL,
+        cl_sync_point_khr* sync_point = NULL,
+        cl_mutable_command_khr* mutable_handle = NULL)
+    {
+        cl_sync_point_khr tmp_sync_point;
+        cl_int error = detail::errHandler(
+            pfn_clCommandCopyImageToBufferKHR(object_,
+                (command_queue != NULL) ? command_queue->operator()() : NULL,
+                src(),
+                dst(),
+                src_origin.data(),
+                region.data(),
+                dst_offset,
+                (sync_points_vec != NULL) ? (cl_uint) sync_points_vec->size() : 0,
+                (sync_points_vec != NULL && sync_points_vec->size() > 0) ? (cl_sync_point_khr*) &sync_points_vec->front() : NULL,
+                (sync_point != NULL) ? &tmp_sync_point : NULL,
+                mutable_handle),
+            __COMMAND_COPY_IMAGE_TO_BUFFER_KHR_ERR);
+
+        if (sync_point != NULL && error == CL_SUCCESS)
+            *sync_point = tmp_sync_point;
+
+        return error;
+    }
+
+    template<typename PatternType>
+    cl_int commandFillBuffer(const CommandQueue* command_queue,
+        const Buffer& buffer,
+        PatternType pattern,
+        size_type offset,
+        size_type size,
+        const vector<cl_sync_point_khr>* sync_points_vec = NULL,
+        cl_sync_point_khr* sync_point = NULL,
+        cl_mutable_command_khr* mutable_handle = NULL)
+    {
+        cl_sync_point_khr tmp_sync_point;
+        cl_int error = detail::errHandler(
+            pfn_clCommandFillBufferKHR(object_,
+                (command_queue != NULL) ? command_queue->operator()() : NULL,
+                buffer(),
+                static_cast<void*>(&pattern),
+                sizeof(PatternType),
+                offset,
+                size,
+                (sync_points_vec != NULL) ? (cl_uint) sync_points_vec->size() : 0,
+                (sync_points_vec != NULL && sync_points_vec->size() > 0) ? (cl_sync_point_khr*) &sync_points_vec->front() : NULL,
+                (sync_point != NULL) ? &tmp_sync_point : NULL,
+                mutable_handle),
+            __COMMAND_FILL_BUFFER_KHR_ERR);
+
+        if (sync_point != NULL && error == CL_SUCCESS)
+            *sync_point = tmp_sync_point;
+
+        return error;
+    }
+
+    cl_int commandFillImage(const CommandQueue* command_queue,
```

Comment:
command_queue could be NULL,so why we can not put it optional parameter?

Reply:
I will fix this for all methods where command_queue could be NULL.